hat/app/org/hatdex/hat/api/controllers/Authentication.scala

```diff
@@ -223,4 +225,100 @@ class Authentication @Inject() (
         Future.successful(Unauthorized(Json.toJson(ErrorMessage("Invalid Token", "Token does not exist"))))
     }
   }
+
+  /*
+    BEGIN: hat-claim
+    Entire Section below is on HAT Claim.
+    TODO: Refactor to New File?
+   */
+  /**
+   * Sends an email to the owner with a link to claim the hat
+   */
+  def claim(applicationId: String): Action[AnyContent] = SecuredAction(WithRole(Owner())).async { implicit request =>
+    val email = request.dynamicEnvironment.ownerEmail
+    val response = Ok(Json.toJson(SuccessResponse("You will shortly receive an email with claim instructions")))
+
+    applicationsService.applicationStatus().flatMap { applications =>
+      applications.find(_.application.id.equals(applicationId)) match {
+        case Some(app) =>
+          usersService.listUsers.map(_.find(_.roles.contains(Owner()))).map {
+            case Some(_) =>
+              val token = MailTokenUser(email, isSignUp = false)
+              tokenService.create(token).map { _ =>
+                val scheme = if (request.secure) {
+                  "https://"
+                }
+                else {
+                  "http://"
+                }
+                val claimLink = s"$scheme${request.host}/#/hat/claim/${token.id}"
+                mailer.claimHat(email, claimLink, app)
```

Comment:
It's unnecessary to send the whole application object to the email handler, partner logo link would suffice.

Reply:
Fixed as requested.